src/components/Semester.vue

```diff
@@ -354,20 +354,17 @@ export default Vue.extend({
     dragListener(event: Event) {
       if (!this.$data.scrollable) event.preventDefault();
     },
-    buildCautions() {
-      this.buildDuplicateCautions();
-    },
-    buildDuplicateCautions() {
-      this.$emit('build-duplicate-cautions');
-    },
+    // TODO: unused
     buildIncorrectPlacementCautions() {
       if (this.courses) {
         this.courses.forEach(course => {
           if (!course.semesters.includes(this.type))
+            // @ts-ignore
             course.alerts.caution = `Course unavailable in the ${this.type}`;
         });
       }
     },
```

Comment:
This is unused for a while. Not sure what's happening... It also appears to me that the author never considered the possibility that a class can both be duplicated and unavailable.